backend/internal/service/handler/spotify.go

```diff
@@ -0,0 +1,149 @@
+package handler
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"math/big"
+	"net/http"
+	"strings"
+
+	spotifyauth "github.com/zmb3/spotify/v2/auth"
+
+	"platnm/internal/config"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/sethvargo/go-envconfig"
+	"github.com/zmb3/spotify/v2"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+var (
```

Comment:
why all this in global vars? would fields on a struct make more sense? the "constructor" of the struct could be responsible for err handling instead of having a global err